src/pivot-table/hooks/use-data-model.ts

```diff
@@ -68,15 +82,53 @@ export default function useDataModel({ model, nextPageHandler, pageInfo }: UseDa
     [genericObjectModel, nextPageHandler, pageInfo, currentPage],
   );
 
+  const applyColumnWidth = useCallback<ApplyColumnWidth>(
+    (newColumnWidth: ColumnWidth, { isPseudoDimension, isAncestorPseudoDimension, isLeftColumn, x, y }: Cell) => {
+      let index: number;
+      if (isPseudoDimension) {
+        // TODO: what todo with the left pseudo dim? set all measures?
+        index = isLeftColumn ? 0 : layoutService.getMeasureInfoIndexFromCellIndex(x);
+      } else {
+        index = isLeftColumn
+          ? x - Number(isAncestorPseudoDimension)
+          : y - Number(isAncestorPseudoDimension) + layoutService.layout.qHyperCube.qNoOfLeftDims;
+      }
+
+      const qPath = `${Q_PATH}/${isPseudoDimension ? "qMeasures" : "qDimensions"}/${index}/qDef/columnWidth`;
+      const oldColumnWidth =
+        layoutService.layout.qHyperCube[isPseudoDimension ? "qMeasureInfo" : "qDimensionInfo"][index].columnWidth;
+      const patch = oldColumnWidth
+        ? {
+            qPath,
+            qOp: "Replace" as EngineAPI.NxPatchOpType,
+            qValue: JSON.stringify({ ...oldColumnWidth, ...newColumnWidth }),
+          }
+        : {
+            qPath,
+            qOp: "Add" as EngineAPI.NxPatchOpType,
+            qValue: JSON.stringify(newColumnWidth),
+          };
+
+      // typescript doesn't like unresolved promises, so we have to do a no-op then
+      // there is nothing that needs to happen after this, so no need for the function to be async
+      model?.applyPatches([patch], true).then(
```

Comment:
I think you can do `await model?.applyPatches([patch], true);` instead

Reply:
That makes this function async, then the function we use for mouseup becomes async, and then I get a type error

Reply:
Odd I did not get the type error when I tested it. Did you get the error in VSC and/or from `yarn types:check`

Reply:
good question, I can try it again

Reply:
it is a lint issue. I could just ignore it on this line then no function would be async